src/main/java/com/pinterest/secor/parser/JsonMessageParser.java

```diff
@@ -36,7 +37,14 @@ public long extractTimestampMillis(final Message message) {
         if (jsonObject != null) {
             Object fieldValue = jsonObject.get(mConfig.getMessageTimestampName());
             if (fieldValue != null) {
-                return toMillis(Double.valueOf(fieldValue.toString()).longValue());
+                String timestamp = fieldValue.toString();
```

Comment:
This functionality (parsing date string in a DateFormatter) is already provided in DatesMessageParser.java